### ISSUE TITLE
RequestSession callback doesn't cope with exceptions thrown from session handler

### ISSUE THREAD

**Issue by dominics** [bugsnag/bugsnag-php#405]
Using Silex as an example below, just because that's what I was using to verify the behavior. Context is that yesterday I deployed a buggy session handler that in some circumstances threw an exception (PDOException specifically).

### Expected behavior

This is the basic set up that is working as expected:

```php
$app = new Silex\Application();

$app->register(new \Silex\Provider\SessionServiceProvider());
$app->register(new \Bugsnag\Silex\Silex2ServiceProvider(), [
    'bugsnag.options' => [
        'api_key' => 'deadbeef',
    ]
]);

$app->error(function (\Exception $e) use ($app) {
    $app['bugsnag']->notifyException($e);
});

$app->get('/', function () {
    throw new \RuntimeException('Some error happened in the controller');
});

return $app;
```

In this case, the exception in the controller is correctly reported. 

### Observed behavior

However, with the addition of a buggy session handler:

```php
$app['session.storage.handler'] = function () {
    return new class extends SessionHandler {
        public function open($save_path, $session_name)
        {
            throw new \RuntimeException('Some error happened in the session handler');
        }
    };
};
```

Now, we get no exception reported at all (neither of the two thrown exceptions makes it to Bugsnag). The stack trace is:

```
Fatal error:  session_start(): Failed to initialize storage module: user (path: ) in /proj/vendor/symfony/http-foundation/Session/Storage/NativeSessionStorage.php on line 144
Stack trace:
  1. {main}() /proj/web/index.php:0
  2. Silex\Application->run() /proj/web/index.php:10
  3. Silex\Application->handle() /proj/vendor/silex/silex/src/Silex/Application.php:477
  4. Symfony\Component\HttpKernel\HttpKernel->handle() /proj/vendor/silex/silex/src/Silex/Application.php:496
  5. Symfony\Component\HttpKernel\HttpKernel->handleException() /proj/vendor/symfony/http-kernel/HttpKernel.php:79
  6. Symfony\Component\EventDispatcher\EventDispatcher->dispatch() /proj/vendor/symfony/http-kernel/HttpKernel.php:230
  7. Symfony\Component\EventDispatcher\EventDispatcher->doDispatch() /proj/vendor/symfony/event-dispatcher/EventDispatcher.php:44
  8. call_user_func:{/proj/vendor/symfony/event-dispatcher/EventDispatcher.php:212}() /proj/vendor/symfony/event-dispatcher/EventDispatcher.php:212
  9. Silex\ExceptionListenerWrapper->__invoke() /proj/vendor/symfony/event-dispatcher/EventDispatcher.php:212
 10. call_user_func:{/proj/vendor/silex/silex/src/Silex/ExceptionListenerWrapper.php:53}() /proj/vendor/silex/silex/src/Silex/ExceptionListenerWrapper.php:53
 11. {closure:/proj/src/app.php:12-14}() /proj/vendor/silex/silex/src/Silex/ExceptionListenerWrapper.php:53
 12. Bugsnag\Client->notifyException() /proj/src/app.php:13
 13. Bugsnag\Client->notify() /proj/vendor/bugsnag/bugsnag/src/Client.php:239
 14. Bugsnag\Pipeline->execute() /proj/vendor/bugsnag/bugsnag/src/Client.php:278
 15. call_user_func:{/proj/vendor/bugsnag/bugsnag/src/Pipeline.php:56}() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:56
 16. Bugsnag\Pipeline->Bugsnag\{closure}() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:56
 17. call_user_func:{/proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68}() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 18. Bugsnag\Middleware\NotificationSkipper->__invoke() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 19. Bugsnag\Pipeline->Bugsnag\{closure}() /proj/vendor/bugsnag/bugsnag/src/Middleware/NotificationSkipper.php:43
 20. call_user_func:{/proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68}() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 21. Bugsnag\Middleware\BreadcrumbData->__invoke() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 22. Bugsnag\Pipeline->Bugsnag\{closure}() /proj/vendor/bugsnag/bugsnag/src/Middleware/BreadcrumbData.php:43
 23. call_user_func:{/proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68}() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 24. Bugsnag\Middleware\CallbackBridge->__invoke() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 25. Bugsnag\Pipeline->Bugsnag\{closure}() /proj/vendor/bugsnag/bugsnag/src/Middleware/CallbackBridge.php:41
 26. call_user_func:{/proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68}() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 27. Bugsnag\Middleware\CallbackBridge->__invoke() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 28. Bugsnag\Pipeline->Bugsnag\{closure}() /proj/vendor/bugsnag/bugsnag/src/Middleware/CallbackBridge.php:41
 29. call_user_func:{/proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68}() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 30. Bugsnag\Middleware\CallbackBridge->__invoke() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 31. Bugsnag\Pipeline->Bugsnag\{closure}() /proj/vendor/bugsnag/bugsnag/src/Middleware/CallbackBridge.php:41
 32. call_user_func:{/proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68}() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 33. Bugsnag\Middleware\CallbackBridge->__invoke() /proj/vendor/bugsnag/bugsnag/src/Pipeline.php:68
 34. Bugsnag\Callbacks\RequestSession->__invoke() /proj/vendor/bugsnag/bugsnag/src/Middleware/CallbackBridge.php:40
 35. Bugsnag\Silex\Request\SilexRequest->getSession() /proj/vendor/bugsnag/bugsnag/src/Callbacks/RequestSession.php:38
 36. Symfony\Component\HttpFoundation\Session\Session->all() /proj/vendor/bugsnag/bugsnag-silex/src/Request/SilexRequest.php:48
 37. Symfony\Component\HttpFoundation\Session\Session->getAttributeBag() /proj/vendor/symfony/http-foundation/Session/Session.php:103
 38. Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage->getBag() /proj/vendor/symfony/http-foundation/Session/Session.php:259
 39. Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage->start() /proj/vendor/symfony/http-foundation/Session/Storage/NativeSessionStorage.php:282
 40. session_start() /proj/vendor/symfony/http-foundation/Session/Storage/NativeSessionStorage.php:144
```

This example also illustrates something annoying about this bug: nothing in the rest of the app accessed the session until Bugsnag did. That means if you're not using the session elsewhere (and you have a buggy handler), Bugsnag can turn a non-fatal `->notifyException()` into a further exception (usually uncaught and fatal because you're already in an exception handler when you notify).

To fix, probably either `Bugsnag\Request\RequestInterface`'s `getSession()` implementations would need to handle exceptions, or the `Bugsnag\Callbacks\RequestSession`'s `__invoke()` implementation would need to. (Or, heck, catch anything thrown during middleware/callback processing and don't let it leak back to the caller?)

### Version

3.6.0

**Comment by dominics** [comment]
Workaround in the meantime is:

* Subclass the request resolver (in my case `Bugsnag\Silex\Request\SilexResolver`), to...
* Return a different `Bugsnag\Request\RequestInterface` implementation (in my case, a subclass of `Bugsnag\Silex\Request`), that...
* Wraps `->getSession()` in a try/catch and returns an empty array if an exception is encountered

**Comment by mattdyoung** [comment]
Fixed in [v2.5.2](https://github.com/bugsnag/bugsnag-silex/releases/tag/v2.5.2)
